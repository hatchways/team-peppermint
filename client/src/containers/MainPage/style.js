import { makeStyles } from "@material-ui/core/styles";

export const useStyles = makeStyles((theme) => ({
  root: {
    // margin: theme.spacing(2),
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    height: "100vh",
    // boxShadow: "0 2px 5px 1px #888",
  },
}));
