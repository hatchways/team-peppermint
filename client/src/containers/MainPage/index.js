import React, { useState } from "react";
import { useStyles } from "./style";
import Sidebar from "../Sidebar";
import MainContentField from "../MainContentField";
import SelectConversation from "../../context/SelectConversation"
const MainPage = () => {
  const classes = useStyles();
  const [conversation, setConversation] = useState()
  return (
    <main className={classes.root}>
      <SelectConversation.Provider value={{ conversation, setConversation }}>
        <Sidebar />
        <MainContentField />
      </SelectConversation.Provider>
    </main>
  );
};

export default MainPage;
