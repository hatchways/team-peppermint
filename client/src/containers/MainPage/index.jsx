import React from "react";
import { useStyles } from "./style";
import Sidebar from "../Sidebar";
import MainContentField from "../MainContentField";

const MainPage = () => {
  const classes = useStyles();
  
  return (
    <main className={classes.root}>
      <Sidebar/>
      <MainContentField/>
    </main>
  );
};

export default MainPage;
