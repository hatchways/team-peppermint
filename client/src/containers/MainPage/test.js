import React from "react";
// import { render } from "@testing-library/react";
import { shallow } from "enzyme";
import MainPage from "./index";

import renderer from "react-test-renderer";

let wrapper;

beforeEach(() => {
  wrapper = shallow(<MainPage />);
});

// test("renders learn react link", () => {
// const { getByText } = render(<App />);
// const linkElement = getByText(/Button/i);
// expect(linkElement).toBeInTheDocument();
// const wrapper = shallow(<App />);
//   const textInButton = wrapper.find("Button").text();

//   expect(textInButton).toBe("Button");
// });

describe("<MainPage />", () => {
  it("match to snapshot", () => {
    // const tree = renderer.create(<MainPage />).toJSON();
    // const wrapper = shallow(<MainPage />);

    expect(wrapper).toMatchSnapshot();
  });
});
