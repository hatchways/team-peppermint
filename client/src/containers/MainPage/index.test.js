import React from "react";
import MainPage from "./index";
import toJSON from "enzyme-to-json";
import { shallow } from "enzyme";

let wrapper;

beforeEach(() => {
  wrapper = shallow(<MainPage />);
});

describe("<MainPage />", () => {
  it("match to snapshot", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
