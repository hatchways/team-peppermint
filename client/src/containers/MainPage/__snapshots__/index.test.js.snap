// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainPage /> match to snapshot 1`] = `
<main
  className="makeStyles-root-1"
>
  <ContextProvider
    value={
      Object {
        "conversation": undefined,
        "setConversation": [Function],
      }
    }
  >
    <ContextProvider
      value={
        Object {
          "original": false,
          "setOriginal": [Function],
        }
      }
    >
      <Sidebar />
      <MainContentField />
    </ContextProvider>
  </ContextProvider>
</main>
`;
